src/uct/ib/mlx5/dv/sig.c

```diff
@@ -266,8 +280,13 @@ ucs_status_t uct_ib_mlx5_sig_mr_init(uct_ib_mlx5_md_t *md,
     unsigned num_elems;
     size_t length;
     void *wqe;
+    int bs;
 
-    if (sig_attr->block != UCS_IB_MLX5_SIGNATURE_BLOCK_4048) {
+    if (sig_attr->block == UCS_IB_MLX5_SIGNATURE_BLOCK_4048) {
```

Comment:
See comment on `#define`s - then this code can be simplified (e.g. access by index)